chore(release): Bump to version 0.5.3

niheaven opened the pull request:
## Changelog

### Features

- **autoupdate:** GitHub predefined hashes support (#6416)

### Bug Fixes

- **scoop-download|install|update:** Fallback to default downloader when aria2 fails (#4292)
- **decompress:** `Expand-7zipArchive` only delete temp dir / `$extractDir` if it is empty (#6092)
- **decompress:** Replace deprecated 7ZIPEXTRACT_USE_EXTERNAL config with USE_EXTERNAL_7ZIP (#6327)
- **commands:** Handling broken aliases (#6141)
- **shim:** Do not suppress `stderr`, properly check `wslpath`/`cygpath` command first (#6114)
- **scoop-bucket:** Add missing import for `no_junction` envs (#6181)
- **scoop-uninstall:** Fix uninstaller does not gain Global state (#6430)
- **scoop-depends-tests:** Mocking `USE_EXTERNAL_7ZIP` as $false to avoding error when it is $true (#6431)

### Code Refactoring

- **download:** Move download-related functions to 'download.ps1' (#6095)
- **Get-Manifest:** Select actual source for manifest (#6142)

### Performance Improvements

- **shim:** Update kiennq-shim to v3.1.2 (#6261)

## Close Issues

- Closes #6011
- Closes #4472
- Closes #6112
- Closes #6180
- Closes #6225
- Closes #6381
- Closes #2719
- Closes #5172
- Closes #5709

## Credits

Thanks to the following contributors (ordered by PR merging time):

@o-l-a-v @niheaven @kiennq @HUMORCE @sitiom @chawyehsu @basselworkforce @WordlessEcho

o-l-a-v committed:
fix (decompress): `Expand-7zipArchive` only delete temp dir / `$extractDir` if it is empty (#6092)

Co-authored-by: Hsiao-nan Cheung <niheaven@gmail.com>

niheaven committed:
refactor(download): Move download-related functions to 'download.ps1' (#6095)

kiennq committed:
fix(download): Fallback to default downloader when aria2 fails (#4292)

HUMORCE committed:
fix(commands): Handling broken aliases (#6141)

sitiom committed:
fix(shim): properly check `wslpath`/`cygpath` command first (#6114)

Co-authored-by: Hsiao-nan Cheung <niheaven@gmail.com>

chawyehsu committed:
fix(scoop-bucket): Add missing import for `no_junction` envs (#6181)

Signed-off-by: Chawye Hsu <su+git@chawyehsu.com>

niheaven committed:
docs(chglog): Update to 0.5.3 (#6258)

niheaven committed:
perf(shim): Update kiennq-shim to v3.1.2 (#6261)

niheaven commented:
@ScoopInstaller/maintainers Anyone could review this PR?

chawyehsu commented on lib/core.ps1:
```diff
@@ -1070,11 +1003,18 @@ function shim($path, $global, $name, $arg) {
+            "  set args=$quoted_arg %*",
+            '  setlocal enabledelayedexpansion',
+            '  if not "!args!"=="" set args=!args:"=""!',
+            "  bash -c `"`$(cygpath -u '$resolved_path') !args!`"",
```
This change on the shim function looks critical, can we have some tests for it?

Everything else seems trivial.

basselworkforce committed:
fix(decompress): Replace deprecated 7ZIPEXTRACT_USE_EXTERNAL config (#6327)

Co-authored-by: Hsiao-nan Cheung <niheaven@gmail.com>

Lutra-Fs commented:
> Hi @ScoopInstaller/maintainers, let's have a poll before we start allowing the use of AI coding tools in the Scoop project.
> 
> 
> 
> Personally I'd be against using them. Bigger projects like curl are already struggling with AI generated bug reports - see [this Ars Technica article](https://arstechnica.com/gadgets/2025/05/open-source-project-curl-is-sick-of-users-submitting-ai-slop-vulnerabilities/) - and [they can't be trusted to write safe code](https://arstechnica.com/security/2025/05/researchers-cause-gitlab-ai-developer-assistant-to-turn-safe-code-malicious/). There have also been a lot of people [wanting the option to turn them off per repository](https://github.com/orgs/community/discussions/159749).
> 
> 
> 
> 👍 if we should use them, 👎  if not!

It’s definitely an important conversation to have while AI is becoming more and more powerful and dangerous.

Personally, I believe AI-assisted tools (especially code review tools) are not inherently bad and can offer useful insights when used responsibly.

As far as I know, GitHub Copilot Agent (which can act more autonomously) is disabled by default, and although GitHub allows per-repo settings, there’s currently no reliable way to enforce those settings on contributors’ machines. Even if Copilot is disabled, users can still access AI support through local tools like Cursor or Cline.

I do agree that AI often lacks full context across files and can suggest speculative issues without proper validation — that’s definitely a concern, especially when it leads to bug reports with no reproduction steps. Those shouldn’t be encouraged.

That said, with good engineering discipline, AI tools have also helped researchers uncover real issues — including a recent example involving Linux kernel bugs: https://news.ycombinator.com/item?id=44081338

In my view, banning AI tools entirely might not be practical or necessary. Instead, we could focus on setting expectations around quality:
	•	no AI-generated PRs without human collaboration (coding,testing) and verification 
	•	no unverifiable bug reports
	•	and encourage responsible, transparent usage.

Lutra-Fs commented:
The AI issue/PR flood is something GitHub has to fight against. I mean, those malicious submits do add a lot of workloads to maintainers, whether or not we do anything/nothing to them; their existence themselves is a workload.

HUMORCE committed:
fix(scoop-uninstall): Fix uninstaller does not gain Global state (#6430)

* global arg

* changelog

HUMORCE committed:
refactor(Get-Manifest): Select actual source for manifest (#6142)

* first step

* Revert "first step"

This reverts commit c5907c3e252982c2fd4f061e87e4b38d251b108c.

* refactor(Get-Manifest): Select actual source for installed manifest

* rework sub-commands, `scoop-depends` is NOT working at this stage

* URI manifest

* opt

* deprecated manifest

* source of manifests

* source of manifest pt2

- Mark URI(path/URL/UNC/etc.) query as standalone manifest
- Drop `installed` and `available update` items for [query] and [installed] are different sources.

* remove variable preventing I forget it

* scoop-info: fix source of manifest on bucket

* fix `scoop-depends`

* Fix Standalone and Source detection

* fix global install

* Fix scoop-cat, scoop-home

- Query for remote manifest

* scoop-list: info +deprecated

* manifest: Fix first selected manifest

* gramma..

* Fix 61b3259

* length

HUMORCE committed:
fix(scoop-depends-tests): Mocking `USE_EXTERNAL_7ZIP` as $false (#6431)

* fix(scoop-depends-tests): Mocking `USE_EXTERNAL_7ZIP` as $false to avoding error when it is $true

* CHANGELOG

WordlessEcho committed:
feat(autoupdate): GitHub predefined hashes support (#6416)

* feat(autoupdate): predefined hash case for GitHub

- Remove `sha256:` prefix in `format_hash()`
- Add GitHub support in `get_hash_for_app()`

Close #6381

* doc(chglog): GitHub auto hash update

* fix(autoupdate): remove prefix only

HUMORCE committed:
docs(CHANGELOG): Update to 0.5.3 (#6432)

* docs(CHANGELOG): Update to 0.5.3

* 6416

WordlessEcho commented:
Also see hotfix #6416 #6436
